flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -56,7 +58,8 @@ abstract class BaseDeltaTaskWriter extends BaseTaskWriter<RowData> {
       Schema schema,
       RowType flinkSchema,
       List<Integer> equalityFieldIds,
-      boolean upsert) {
+      boolean upsert,
+      DeleteGranularity deleteGranularity) {
```

Comment:
Is there a use case where we want to use partition granularity? Based on our initial discussions, we wanted to always use file granularity in Flink.

Reply:
I think that in most cases we want to use FILE granularity, but we found a case when Flink might ending up writing multiple delete files with FILE granularity - when there are big writes, and the data writer rolls. In this case we might have to create multiple delete files for a single commit. This might be undesirable for some users, so I have kept the option for now.

Reply:
Are we worried about the increased metadata size?